edgelet/edgelet-core/src/watchdog.rs

```diff
@@ -26,9 +28,32 @@ const MODULE_GENERATIONID: &str = "IOTEDGE_MODULEGENERATIONID";
 /// This is the frequency with which the watchdog checks for the status of the edge runtime module.
 const WATCHDOG_FREQUENCY_SECS: u64 = 60;
 
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum RetryLimit {
+    Infinite,
+    Num(u32),
```

Comment:
It's not going to matter, but countable things are usually measured in `usize` rather than `u32`.